lib/folio_record.rb

```diff
@@ -80,12 +80,35 @@ def sirsi_holdings
         current_location: (current_location unless current_location == home_location_code),
         home_location: home_location_code,
         library: library_code,
-        scheme: call_number_type_map(item.dig('callNumber', 'typeName')),
+        scheme: call_number_type_map(item.dig('callNumberType', 'name')),
         type: item['materialType'],
         barcode: item['barcode'],
         public_note: item['notes']&.map { |n| ".#{n['itemNoteTypeName']&.upcase}. #{n['note']}" }&.join("\n"),
         tag: item
       )
+    end.concat(bound_with_holdings)
+  end
+
+  # since FOLIO Bound-with records don't have items, we generate a SirsiHolding using data from the parent item and child holding
+  # without doing this and appending above, SearchWorks does not display the Bound-with record properly
+  # TODO: remove this when we stop using SirsiHoldings
+  def bound_with_holdings
+    @bound_with_holdings ||= holdings.filter { |holding| holding.dig('holdingsType', 'name') == 'Bound-with' }.map do |holding|
+      parent_item = record['boundWithParents'].find { |parent| parent['childHoldingId'] == holding['id'] }
+      parent_item_location_code = parent_item.dig('parentItemLocation', 'permanentLocation', 'code')
+      library_code, home_location_code = LocationsMap.for(parent_item_location_code)
+      _current_library, current_location = LocationsMap.for(parent_item.dig('parentItemLocation', 'effectiveLocation', 'code'))
+      SirsiHolding.new(
+        call_number: holding['callNumber'],
+        current_location:,
+        home_location: home_location_code,
+        library: library_code,
+        scheme: call_number_type_map(holding.dig('callNumberType', 'name')),
+        type: '',
+        barcode: '',
+        public_note: '',
+        tag: ''
```

Comment:
what's the impact of not including these (especially barcode)? won't we need to render an item-level request link for boundwiths, which would need a barcode?

Reply:
Good point, though I wasn't able to make much progress on the request button without circ rules. For this ticket trying to prevent "On Order" status felt like an actionable chunk. But I'm happy to try to populate these fields here if that makes most sense.

Reply:
I feel like we may as well use these fields from the child, and see what happens? At least for testing, it might be good for the time being.

This is one of those cases where I think trying to get parity is a _bad_ idea; I think we saw that in current production, you get a location-level request link for the parent...which would make a librarian have to do the work to connect them. So, let's try something new here and see what happens.

Reply:
Sounds good. For me this raised questions about naming ("parent_barcode"?), SW design, etc., that were beyond what's needed to satisfy this particular ticket about preventing the On Order status.

Reply:
That's fair — I think the underlying question of the ticket is "if not on-order, how should they appear?", and the important corollary is "they're not going to appear the same as how they do in current production". Preventing on-order status means that they still need to show as something else, so I think we should go for the more predictable thing here and not leave fields blank.

Reply:
What would parity be? I think we saw that in existing symphony records, bound-with children have items, but in FOLIO they don't. It doesn't seem like there is a way to achieve parity in that situation.

Reply:
@marlo-longley I guess we just need to go back and look at the `item_display` field for a bound-with in current production and double-check where each of the values comes from, and then try to recreate that. it might be possible...